crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -180,48 +180,56 @@ impl Solution {
     /// Check that the sum of tokens entering the settlement is not less than
     /// the sum of tokens exiting the settlement.
     fn verify_asset_flow(&self) -> Result<(), VerificationError> {
```

Comment:
This still seems wrong to me. Asset flow verification should be done on individual token amounts and not on estimated reference amounts (which is not always available).